runtime/lua/vim/diagnostic.lua

```diff
@@ -189,8 +189,48 @@ end
 
 --- @class vim.diagnostic.Opts.Status
 ---
---- A table mapping |diagnostic-severity| to the text to use for each severity section.
---- @field text? table<vim.diagnostic.Severity,string>
+--- Only show diagnostics matching the given severity |diagnostic-severity|.
+--- @field severity? vim.diagnostic.SeverityFilter
```

Comment:
Can we remove the `severity` field here for the sake of keeping this PR exclusive to the `format` option? We can add this setting later.

Reply:
Yeah, you're right